src/App.tsx

```diff
@@ -15,36 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortBy, isReversed }: SortParams) {
+  const preparedGoods = [...goods];
+
+  if (sortBy !== SortType.default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
```

Comment:
[CODE KNOWLEDGE] - The function `getPreparedGoods` interacts with the list of goods, which is part of the component's state. Consider moving this logic inside the component or use a custom hook to encapsulate the sorting logic, so it can benefit from React's state management and avoid direct manipulation of data.